test/Microsoft.AspNet.Mvc.ModelBinding.Test/Validation/ModelStateDictionaryTest.cs

```diff
@@ -733,6 +734,53 @@ public void ModelStateDictionary_NoErrorMessage_ForNonFormatException()
             Assert.Empty(error.ErrorMessage);
         }
 
+        [Fact]
+        public void ModelStateDictionary_ClearEntriesPrefixedWithKey_NonEmptyKey()
```

Comment:
This test doesn't really handle `Prefix` it handles exact matches. Would like to see one that deals with elements and sub-properties. 

Ex:
`Property1` clears `Property1.Foo[0]`, `Property1.Bar`